### PR TITLE
Leverage new subscriber image endpoints in external-solution test

### PR DESCRIPTION
**Pull request by grischperl**
**Description**

Changes proposed in this pull request:

- Changed the function to the cluster
   function now POSTS to the endpoint which stores the received CloudEvent
- Corrected event-subscriber readme to now include the correct endpoint to GET the CloudEvent with a matching UUID
- Combined the steps for sending and checking the event ID to allow a counter which differentiates the events for each retry
- Implemented checking the event by ID
- Implemented returning a list of all received events if correct event hasn't been received

**Related issue(s)**

#8599

### CODE REVIEW

File: tests/end-to-end/external-solution-integration/pkg/testsuite/check_event_id.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+// NewCheckEventId returns new CheckEventId
+func NewCheckEventId(testService *testkit.TestService, eventId string, opts ...retrygo.Option) *CheckEventId {
+	return &CheckEventId{
+		testService: testService,
+		eventId:     eventId,
+		retryOpts:   opts,
+	}
+}
+
+// Name returns name name of the step
```
Comment (k15r) [inline tests/end-to-end/external-solution-integration/pkg/testsuite/check_event_id.go:28]:
```suggestion
// Name returns name of the step
```

File: tests/end-to-end/external-solution-integration/pkg/testsuite/check_event_id.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+// Run executes the step
+func (s *CheckEventId) Run() error {
+	err := retry.Do(func() error {
+		return s.testService.CheckEventId(s.eventId)
+	}, s.retryOpts...)
+	if err != nil {
+		//if all retries fail, the test should ask the subscriber for all events which it received
+		return s.testService.CheckAllReceivedEvents()
+		//return errors.Wrapf(err, "the event id could not be checked")
```
Comment (k15r) [inline tests/end-to-end/external-solution-integration/pkg/testsuite/check_event_id.go:41]: dead code should be removed

File: tests/end-to-end/external-solution-integration/pkg/testsuite/deploy_function.go [hunk 3]
```diff
@@ -32,8 +32,17 @@ function rejected(result) {
   console.log('Rejected:');	
   console.log(JSON.stringify(result, null, 2));	
 }
-function sendReq(url, resolve, reject) {	
-    request.post(url, { json: true }, (error, response, body) => {	
+function sendReq(event, url, resolve, reject) {
```
Comment (k15r) [inline tests/end-to-end/external-solution-integration/pkg/testsuite/deploy_function.go:35]: please reindent this js snippet, it makes it easier to read

Comment (grischperl) [reply tests/end-to-end/external-solution-integration/pkg/testsuite/deploy_function.go:35]: Fixed

File: tests/end-to-end/external-solution-integration/pkg/testsuite/check_event_id.go (hunk 2 above)
Comment (grischperl) [reply tests/end-to-end/external-solution-integration/pkg/testsuite/check_event_id.go]: Fixed

File: tests/end-to-end/external-solution-integration/pkg/testsuite/check_event_id.go (hunk 1 above)
Comment (grischperl) [reply tests/end-to-end/external-solution-integration/pkg/testsuite/check_event_id.go]: Fixed

File: tests/end-to-end/external-solution-integration/pkg/testsuite/check_event_id.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+import (
+	retrygo "github.com/avast/retry-go"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/retry"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/step"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/pkg/testkit"
+)
+
+// CheckEventId is a step which checks if the correct EventId has been received
+type CheckEventId struct {
```
Comment (marcobebway) [inline tests/end-to-end/external-solution-integration/pkg/testsuite/check_event_id.go:11]: I could not find where is this used?

File: tests/end-to-end/external-solution-integration/pkg/testsuite/send_event_to_compatibility_layer_and_check_eventId.go [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+// SendEventToCompatibilityLayerAndCheckEventId is a step which sends an event and checks if the correct EventId has been received
+type SendEventToCompatibilityLayerAndCheckEventId struct {
+	state       SendEventState
+	appName     string
+	payload     string
+	testService *testkit.TestService
+	retryOpts   []retrygo.Option
+}
+
+var counterLayer = 0
```
Comment (marcobebway) [inline tests/end-to-end/external-solution-integration/pkg/testsuite/send_event_to_compatibility_layer_and_check_eventId.go:24]: Does it make sense to move this inside the `SendEventToCompatibilityLayerAndCheckEventId` struct definition?

File: tests/end-to-end/external-solution-integration/pkg/testsuite/send_event_to_mesh_and_check_eventId.go [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+// SendEventToMeshAndCheckEventId is a step which sends an event and checks if the correct EventId has been received
+type SendEventToMeshAndCheckEventId struct {
+	state       SendEventState
+	appName     string
+	payload     string
+	testService *testkit.TestService
+	retryOpts   []retrygo.Option
+}
+
+var counterMesh = 0
```
Comment (marcobebway) [inline tests/end-to-end/external-solution-integration/pkg/testsuite/send_event_to_mesh_and_check_eventId.go:27]: Does it make sense to move this inside the `SendEventToMeshAndCheckEventId` struct definition?

**Review by alexandra-simeonova — DISMISSED** [review]

File: tests/end-to-end/external-solution-integration/pkg/testkit/testservice.go [hunk 7]
```diff
@@ -103,6 +105,67 @@ func (ts *TestService) checkValue() (int, error) {
 	return response.Counter, nil
 }
 
+func (ts *TestService) CheckAllReceivedEvents() error {
```
Comment (marcobebway) [inline tests/end-to-end/external-solution-integration/pkg/testkit/testservice.go:108]: Does it make sense to rename the function to something like `DumpAllReceivedEvents`?

Comment (grischperl) [reply tests/end-to-end/external-solution-integration/pkg/testkit/testservice.go]: Yes, renamed the function and added documentation for clarity

File: tests/end-to-end/external-solution-integration/pkg/testsuite/send_event_to_mesh_and_check_eventId.go (hunk 6 above)
Comment (grischperl) [reply tests/end-to-end/external-solution-integration/pkg/testsuite/send_event_to_mesh_and_check_eventId.go]: Fixed

File: tests/end-to-end/external-solution-integration/pkg/testsuite/send_event_to_compatibility_layer_and_check_eventId.go (hunk 5 above)
Comment (grischperl) [reply tests/end-to-end/external-solution-integration/pkg/testsuite/send_event_to_compatibility_layer_and_check_eventId.go]: Fixed

File: tests/end-to-end/external-solution-integration/pkg/testsuite/check_event_id.go (hunk 4 above)
Comment (grischperl) [reply tests/end-to-end/external-solution-integration/pkg/testsuite/check_event_id.go]: Yes, this is a remainder from before the steps where combined and should have been removed! Fixed it now

**Review by marcobebway — DISMISSED** [review]

**Review by marcobebway — APPROVED** [review]

**Review by k15r — APPROVED** [review]

**Review by alexandra-simeonova — APPROVED** [review]

**Pull request merged**